src/Geometry/Vector2.cpp

```diff
@@ -2,6 +2,12 @@
 
 using namespace Geometry;
 
+Vector2<double> Geometry::operator+(const Vector2<double> &left, const Vector2<float> &right){
+	return left + Vector2<double>(right);
+}
+Vector2<double> Geometry::operator+(const Vector2<float> &left, const Vector2<double> &right){
+	return Vector2<double>(left) + right;
```

Comment:
```suggestion
	return right + Vector2<double>(left);
```
Uses associativity of addition here. In that case, if you change something in one of the two, you don't have to change the other one.